pkg/tenant/tenants.go

```diff
@@ -0,0 +1,76 @@
+package tenant
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
+	"github.com/rotationalio/ensign/pkg/tenant/db"
+	"github.com/rs/zerolog/log"
+)
+
+// TenantCreates creates adds a new tenant to the database
+func (s *Server) TenantCreate(c *gin.Context) {
+	var (
+		err error
+		t   *api.Tenant
+	)
+
+	// TODO: Add authentication and authorization middleware
+
+	if err = c.BindJSON(&t); err != nil {
+		log.Warn().Err(err).Msg("could not bind tenant create request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
+	}
+
+	if t.ID != "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrTenantIDRequired))
```

Comment:
Shouldn't this error be something like "cannot specify tenant ID on create" rather than "tenant ID required"?

Reply:
I'll update the error to be more accurate. After all making changes to find the error, I didn't properly update this.